### PR TITLE
fix issue #85

### PR DESCRIPTION
**Pull request by kalavt**
Handle keys with dashes/dots by replacing them with underlines

### CODE REVIEW

File: dotmap/__init__.py [hunk 1]
```diff
@@ -39,6 +39,7 @@ def __init__(self, *args, **kwargs):
             for k,v in src:
                 if self._prevent_method_masking and k in reserved_keys:
                     raise KeyError('"{}" is reserved'.format(k))
+                k = self.__fix_ilegal_name__(k)
```
Comment (drgrib) [inline dotmap/__init__.py:42]: Can you rename this to `__fix_illegal_name__`? Specifically, `ilegal` is spelled incorrectly.

**Comment by drgrib** [comment]
Have you already ran testing and gotten back no issues?
```
python3 -m unittest
```

**Comment by kalavt** [comment]
Yes, 
`
$ python3 -m unittest
............................
----------------------------------------------------------------------
Ran 28 tests in 0.019s

OK
`

File: dotmap/__init__.py (hunk 1 above)
Comment (kalavt) [reply dotmap/__init__.py]: sorry for the typo, fixed.

**Comment by drgrib** [comment]
Awesome. Thanks for writing this.

**Pull request merged**